lib/rights_api/schema.rb

```diff
@@ -25,39 +26,59 @@ class Schema
       rights_log: :rights_log,
       sources: :sources
     }.freeze
-    private_constant :NAME_TO_TABLE
+
+    EXPANSIONS = {
+      "attr" => "attribute",
+      "dscr" => "description",
+      "stmt" => "statement"
+    }.freeze
+
+    private_constant :NAME_TO_TABLE, :EXPANSIONS
 
     attr_reader :table
 
     def self.names
       NAME_TO_TABLE.keys.sort
     end
 
+    # @param name [String, Symbol] A table name
     def self.named(name:)
-      case name
-      when :rights, :rights_log
-        RightsSchema
+      case name.to_sym
       when :access_statements
         AccessStatementsSchema
       when :access_statements_map
         AccessStatementsMapSchema
+      when :rights, :rights_log
+        RightsSchema
       else
         Schema
       end.new(table: NAME_TO_TABLE[name])
     end
 
+    # @param table [String, Symbol]
     def initialize(table:)
       @table = table.to_sym
     end
 
-    # @param [Hash] row
     # It should be fine to modify the row in-place.
     # Add, delete, modify the row data to make it conform to
     # the expected structure, in particular:
     # - Delete fields that should not be exposed in a public API.
     # - Rename fields that derive from opaque, abbreviated, or oddball column names.
     # - Add derivative fields with foreign key URLs.
-    def normalize_row(row)
```

Comment:
I get what you're going for here, but I wonder if it might be both more readable and more flexible long-term just to explicitly map the input row to the desired output structure for each table -- that is, build up an output structure given the input record, rather than trying to transform the input record. In particular, while this approach is pretty tolerant to changes to the existing schema, I'm concerned that it might be tying the output structure too closely to the input schema in ways that could be a problem later.

An approach I might consider would be:
* one class for each database table
* some of the existing query methods like `primary_key`, `query_for_field`, etc could be class methods
* an initializer that takes a row from the database and transforms it to some internal representation
* a `to_json` method that outputs an external representation

I'd recommend for the time being explicitly enumerating all the fields in both the input row & the output object, and tolerating some duplication / boilerplate in those initializers & `to_json` method - this then gives us the flexibility to start changing those independently later.

Reply:
I did not go so far as to implement a `to_json` since we build up the result row-by-row, I found that `to_h` fit the bill nicely at least for now. (We may decide to do the JSON ourselves but for now I make Sinatra do it for me.)

For all the boilerplate I feel like I have a better sense of where this is all going, I even tried a little (very inefficient) experiment with returning an attribute structure in a rights query result. `Schema` has been showing "pool of glue" aspirations since I wrote the initial version; this appears to be staving that off.